### DISCUSSION TITLE
RADXA quad SATA HAT support

### DISCUSSION THREAD

**Discussion by rholze** [MichaIng/DietPi#6538] [Software Request]
Hello,
I have a quad SATA HAT from radxa (https://wiki.radxa.com/Dual_Quad_SATA_HAT) for my local cloud-Server. With them there is a script with which you can also set up all existing components. The script is unfortunately only for Buster and not for example Bullseye. I have now found the following project on the Internet: https://github.com/nolltre/rockpi_sata. But unfortunately it is not finished. I wonder now, if it could not be of general interest to integrate a support here also in dietpi-software? I have to set up the drivers again with every update/new installation. In addition, I read that the original implementation was too complex. Unfortunately, I do not have enough expertiese from the topic to put it all myself or even for the dietpi distribution to adjust. I would be happy to talk to someone about this and possibly find a solution.
Thanks a lot
Ronald

**Comment by MichaIng** [comment]
Which SBC you want to use this with? The script they provide pulls other scripts from various places of the web, depending on SBC and distro.

For ROCK 4: https://cos.setf.me/rockpi/sh/penta-sata/debian.sh
For RPi, but probably only compatible with Raspbian (not Debian), but need to have a look: https://s3.setq.io/rockpi/sh/quad-sata/raspbian.sh

The ROCK 4 script works only with their vendor image, not with ours or any other, because it tries to update their kernel and bootloader. And it runs a script to configure I2C, which also won't work with mainline kernel.

... the RPi script should actually work on any RPi with Raspbian or Debian basis (hence all our images), as long as this Adafruit Python project installed is compatible with recent Python 3 versions.
